lib/file-source/src/file_server.rs

```diff
@@ -282,10 +288,12 @@ impl FileServer {
             }
             let backoff = backoff_cap.saturating_sub(global_bytes_read);
 
-            match shutdown.recv_timeout(time::Duration::from_millis(backoff as u64)) {
-                Ok(()) => unreachable!(), // The sender should never actually send
-                Err(RecvTimeoutError::Timeout) => {}
-                Err(RecvTimeoutError::Disconnected) => return,
+            match block_on(select(
```

Comment:
This should panic, so I don't see how this works?

Reply:
You are right it was panicking before for that reason, that's why the whole `file_server` is now executed with `spawn_blocking` instead of `thread_spawn`. https://github.com/timberio/vector/blob/bc2a2c1cfbab9ea45bd9c67f513c81ad85d39d14/src/sources/file/mod.rs#L304

Reply:
Ok, can we add a comment above then why this does work and something that should make it easy to understand why this might break in the future if we no longer use spawn_blocking.